vm/src/vm.cpp

```diff
@@ -1,18 +1,23 @@
 #include "byte_reader.h"
+#include "definitions/module_definition.h"
 #include "interpreter.h"
-#include "memory/constant_pool.h"
-#include "memory/strings.h"
+#include <cxxabi.h>
+
 #include <iostream>
 
 extern "C" void moshell_exec(const char *bytes, size_t byte_count) {
     strings_t strings;
-
+    ByteReader reader(bytes, byte_count);
     try {
-        ByteReader reader(bytes, byte_count);
-        ConstantPool pool = load_constant_pool(reader, strings);
+        // read function definitions
+        auto module_def = load_module(reader, strings);
+
+        run_module(module_def, strings);
 
-        run(pool, bytes + reader.position(), byte_count - reader.position(), strings);
     } catch (const std::exception &e) {
-        std::cerr << e.what() << '\n';
+        int status;
+        char *exception_type = abi::__cxa_demangle(typeid(e).name(), nullptr, 0, &status);
+        std::cerr << exception_type << ": " << e.what() << std::endl;
+        free(exception_type);
```

Comment:
```suggestion
        std::cerr << e.what() << '\n';
```
Don't depend on libstdc++ internals.